src/Internal/Service/PaymentProcessingService.php

```diff
@@ -68,8 +80,10 @@ public function process_payment( int $order_id, bool $manual_capture = false ) {
 
 		$request       = new PaymentRequest( $this->legacy_proxy );
 		$initial_state = $this->state_factory->create_state( InitialState::class, $context );
+		$final_state   = $initial_state->start_processing( $request );
 
-		return $initial_state->start_processing( $request );
+		$this->context_logger_service->log_changes( $context );
```

Comment:
One thing to consider here is that if an exception is thrown within the state machine, this method will not be called. 

So I think we will need to handle that too. For example, catch the exception, log the exception error too, throw the exception again.

Reply:
Yes, I have this in mind. It will be handled as part of the Failures stream of work.